engine/src/main/java/org/pentaho/di/www/GetJobImageServlet.java

```diff
@@ -136,14 +139,20 @@ public void doGet( HttpServletRequest request, HttpServletResponse response ) th
     String jobName = request.getParameter( "name" );
     String id = request.getParameter( "id" );
 
+    boolean useXML = "Y".equalsIgnoreCase( request.getParameter( "xml" ) );
+
     // ID is optional...
     //
     Job job;
-    CarteObjectEntry entry;
+    CarteObjectEntry entry = null;
     if ( Utils.isEmpty( id ) ) {
-      // get the first transformation that matches...
-      //
-      entry = getJobMap().getFirstCarteObjectEntry( jobName );
+
```

Comment:
Can I see a unit test to exercise the new logic?

Reply:
Made several unit tests for GetJobImageServlet